docs/style-guidelines.md

```diff
@@ -17,4 +17,96 @@ To install it go to the root of your explorer repository and run the following c
+        
+        public void Play(float speed)                   // Method -> PascalCase. Method parameters -> camelCase
+        {
+            var animationClip;                          // local variable -> camelCase
+        };              
+    }
+}
+```
+Suggestions:
+* `Interface` - try to name it with adjective phrases (`IDamageable`). If it is difficult, then use descriptive noun (`IBaseVariable`) or noun phrase (`IAssetProvider`).
+* `Class`/`Struct` -  name with nouns (`Health`) or noun phrases (`InputService`).
+
+### Ordering conventions
+* `using` go first and placed alphabetically
+* Class members grouped and appears in the following order
+  * Enums, delegates declarations
+  * Fields
+  * Properties
+  * Events
+  * Methods
+  * Nested classes
+* Order inside group:
+  * `public`
+  * `internal`
+  * `protected internal`
+  * `protected`
+  * `private`
+* Fields specifics
+  * Const, static and readonly goes first
+  * Public `[HideInInspector]` and `[NonSerialized]` attribute goes after `public`
+  * `[SerializedFields]` attribute fields goes after all `public`
+* Properties specifics
+  * static and readonly goes first
+  * Property with public `set` and private `get` goes before get-only Property or Property with private `set` and public `get` (considered to be more exposed)
+```csharp
+public const string ASSET_PATH = "AvatarPrefab";                            // Constants group 
+internal const float MIN_TIME = 0.1f;
+private const float MAX_TIMER = 10f;
+
+public static readonly int WaveAnimHash = Animator.StringToHash("Wave");    // Static Read-only group
+private static readonly int danceAnimHash = Animator.StringToHash("Dance");
+
+internal static bool isOpen;                                                // Static group
+protected static bool isAvtive;                                             
+
+public readonly List<int> CachedNumbers = new List<int>();                  // Read-only group
+private readonly List<Section> Sections = new List<Section>();
+
+public int PublicFieldA;                                                    // Public Fields group
+
+[HideInInspector] public int SomeField02;
+[NonSerialized] public int SomeField03;
+
+[SerializedField] internal Animator animator;                               // [SerializedField]'s group
+[SerializedField] private AnimatorClip[] clips[];  
+
+protected float cooldown;                                                   // internal-protected-private Fields group  
+private bool isVisible;                         
+                       
+public bool Property1 { get; set; }                                        // Properties groups  
+public bool Property2 { private get; set; }
+public bool Property3 { get; }
+public bool Property4 { get; private set; }
+
+protected bool Property5 { get; set; }
+private bool Property6 { get; set; }
+
+public event Action Started;                                               // Events group  
```

Comment:
i would add a prefix on events: `OnStarted`, `OnCompleted`

Reply:
I am more of a fan of a classical Microsoft .NET notation for event naming, where the event itself means finished actions (`Started`, `Changed`)  and the subscriber has an `On` prefix. This is suggested [in Microsoft guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members) 
![image](https://user-images.githubusercontent.com/35366872/191728764-cb32d7a0-4c7d-49d0-853a-f70f1ce25f0d.png)
and also in the Unity code-style guidelines ([blog](https://blog.unity.com/technology/clean-up-your-code-how-to-create-your-own-c-code-style), book(file:///C:/Users/Vitaly/Downloads/create-a-c-style-guide-write-cleaner-code-that-scales%20(1).pdf)), namely saying

> Choose a name that communicates the state change accurately. Use the present or past participle to indicate events “before” or “after.” For example, specify “OpeningDoor” for an event before opening a door or “DoorOpened” for an event afterward.

> Prefix the event raising method (in the subject) with “On”: The subject that invokes the event typically does so from a method prefixed with “On,” e.g. “OnOpeningDoor” or “OnDoorOpened.

Reply:
@Antivortex, mentioned above .NET style (without On prefix), removes the need of having postfixes for subscribing method (like OnSomethingStarted**Handler**), because prefix now plays this role and identifies subscribed methods.

Reply:
i think the prefixes helps to avoid overlapping names with methods or properties. Its also recommended by default on rider

Reply:
It could be that the .NET notation is also related to a readability question, so "code reads like well-written prose" (Grady Booch).` Door.Opened` is more grammatically correct than `Door.OnOpened`, while `OnDoorOpened` also correct and could be used for a subscriber

Reply:
and Event will not overlap with the method name, because Method name should be a verb in the present tense, while event is in the Past tense.

Reply:
To be honest I've always seen them as `OnSomething`, I'd go with Nico's proposal.

Reply:
Then let's go with `OnSomething` 😉